projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.native.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { SandBox } from '@wordpress/components';
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { View, Text } from 'react-native';
+
+/**
+ * VideoPlayer react component
+ *
+ * @param {object} props                 - Component props.
+ * @param {object} props.html            - Player html to render in the sandbox.
+ * @param {object} props.attributes      - Block attributes.
+ * @param {Array} props.scripts          - Scripts to pass trough to the sandbox.
+ * @param {object} props.preview         - oEmbed preview data.
+ * @param {boolean} props.isRequestingEmbedPreview - oEmbed preview data.
+ * @returns {object}                     - React component.
+ */
+export default function VideoPressPlayer( {
+	html,
+	attributes,
+	scripts = [],
+	preview,
+	isRequestingEmbedPreview,
+} ) {
+	const [ videoPlayerHeight, setVideoPlayerHeight ] = useState( 250 );
```

Comment:
[In the web version, a temporary height is set](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L66-L73), maybe we could replicate a similar logic instead of having a constant, WDYT?

Reply:
Are you referencing the magic number `250` ?  That's just a replacement for the temporary styling height that was added [here](https://github.com/Automattic/jetpack/pull/29165/files#diff-a48bfa252db3a7399b5a161b6b36a5b88c3db3d37780daece46693f418238647L12). I would expect that to be modified later once we get into refining the component.  That said the web version [also uses a constant](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L189), just not in the `useState` call. ( Just noticed while writing this that the web version now uses [a constant as well](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L71) )

Or are you talking the web version use of  `state.videoPlayerTemporaryHeight` . It's really just naming.  I've simplified the logic involved in calling `setVideoPlayerHeight` vs the web version `setVideoPlayerTemporaryHeight` but the intent is the same. I expect the calling logic to become more complex in future changes so I opted to keep it simple for now. 
I found the use of _temporary_ misleading in the web version. I couldn't find a condition where the "temporary" height is not used, it is ignored once the embed is rendered but it's not replaced or removed. Perhaps a better name, in both cases, is `playerWrapperHeight` since that is where how the state property is [used](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L201).

Reply:
> Are you referencing the magic number 250 ? That's just a replacement for the temporary styling height that was added [here](https://github.com/Automattic/jetpack/pull/29165/files#diff-a48bfa252db3a7399b5a161b6b36a5b88c3db3d37780daece46693f418238647L12). I would expect that to be modified later once we get into refining the component. That said the web version [also uses a constant](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L189), just not in the useState call. ( Just noticed while writing this that the web version now uses [a constant as well](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L71) )

Yeah, I was referencing the magic number. Oh, true, in the web version we also have a constant. In that case, omit my comment then, thanks for elaborating!

> Or are you talking the web version use of `state.videoPlayerTemporaryHeight` . It's really just naming. I've simplified the logic involved in calling `setVideoPlayerHeight` vs the web version `setVideoPlayerTemporaryHeight` but the intent is the same. I expect the calling logic to become more complex in future changes so I opted to keep it simple for now.
> I found the use of _temporary_ misleading in the web version. I couldn't find a condition where the "temporary" height is not used, it is ignored once the embed is rendered but it's not replaced or removed. Perhaps a better name, in both cases, is `playerWrapperHeight` since that is where how the state property is [used](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L201).

Based on the following comment, I understand that the temporary height is just an initial calculation we make based on the block's width and the video's aspect ratio. Once the player is loaded, the height is replaced with the final value, so it seems the temporariness is based on the fact that is only used during the loading state.

https://github.com/Automattic/jetpack/blob/8f7114bde33077e8b45940ac579126c4d77a2ac8/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L60-L66

Reply:
It's true that there is an initial value for the height that is derived from the aspect ratio. 
The issue I see is the naming. From what I observed the `videoPlayerTemporaryHeight` state variable is always used to set the style height property. 

In fact once the video is available, 'videoPlayerTemporaryHeight` is set to `auto` [here](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L103) 

In other words, this state variable has an initial value to set a placeholder height, and then is set to `auto` once the actual height is available. To me that variable is not temporary. If I see something named temporary with a comment that it's eventually ignored,  I would expect to see it used like 

```
const value = actualValue || temporaryValue
```

instead the code here is more like 
```
let temporaryValue = 1;
eventually( (actualValue) => setTemporaryValue(actualValue) )
const value = temporaryValue
```